packages/test/end-to-end-tests/src/test/container.spec.ts

```diff
@@ -181,6 +183,27 @@ describe("Container", () => {
         deltaConnection.removeAllListeners();
     });
 
+    it("Delta manager receives readonly event when calling container.forceReadonly()", async () => {
```

Comment:
I don't think this test belongs here. This file is testing Container loading / connection and events around them. The name of the file is misleading :-).
I would consider adding it to a new test file that tests delta manager functionality or tests readonly functionality.

Reply:
I created a bunch of unit tests for delta manager itself. @vladsud suggested I also add an end to end test that validates, when we call forceReadonly() on the container, that the event can be observed. I thought that this would be a good place, since it is a container API we are testing in the container test suite.

I don't see much value in adding a new delta manager test file since we already unit test this behavior for the delta manager itself.

Reply:
Makes sense. Thanks for the background on this! Hence my other question as to what we are trying to test here :-).

Can you also add a check that container.readonly has changed to true after setting forceReadonly? That would make sure that we don't break it.

Reply:
Sure, I'll add.

Reply:
Thanks Vlad!
I think eventually we would want more readonly mode tests and we can restructure this. For now, we should create an issue that tracks adding more tests.

Reply:
Created #4421 to track this.